### PR TITLE
#535 Add transaction command to remove non-root certificates

### PR DESCRIPTION
**Pull request by Abdulbois**

### CODE REVIEW

File: integration_tests/cli/pki-remove-x509-certificates.sh [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+check_response "$result" "\"subject\": \"$intermediate_cert_subject\""
+check_response "$result" "\"subjectKeyId\": \"$root_cert_subject_key_id\""
+check_response "$result" "\"subjectKeyId\": \"$intermediate_cert_subject_key_id\""
+check_response "$result" "\"subjectKeyId\": \"$leaf_cert_subject_key_id\""
+check_response "$result" "\"serialNumber\": \"$root_cert_1_serial_number\""
+check_response "$result" "\"serialNumber\": \"$intermediate_cert_2_serial_number\""
+check_response "$result" "\"serialNumber\": \"$leaf_cert_serial_number\""
+response_does_not_contain "$result" "\"serialNumber\": \"$intermediate_cert_1_serial_number\""
+
+echo "Request all approved intermediate certificates should contain only one certificate with serialNumber 4"
```
Comment (akarabashov) [inline integration_tests/cli/pki-remove-x509-certificates.sh:79]: Request approved certificates by an intermediate certificate's subject and subjectKeyId should return a single certificate with serialNumber 4

File: integration_tests/cli/pki-remove-x509-certificates.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+check_response "$result" "\"subject\": \"$intermediate_cert_subject\""
+check_response "$result" "\"subjectKeyId\": \"$intermediate_cert_subject_key_id\""
+check_response "$result" "\"serialNumber\": \"$intermediate_cert_2_serial_number\""
+response_does_not_contain "$result" "\"serialNumber\": \"$intermediate_cert_1_serial_number\""
+
+echo "Remove intermediate certificate with subject and subjectKeyId"
+result=$(echo "$passphrase" | dcld tx pki remove-x509-cert --subject="$intermediate_cert_subject" --subject-key-id="$intermediate_cert_subject_key_id" --from=$trustee_account --yes)
+check_response "$result" "\"code\": 0"
+
+echo "Request approved intermediate certificates should be empty"
```
Comment (akarabashov) [inline integration_tests/cli/pki-remove-x509-certificates.sh:91]: Request approved certificates by an intermediate certificate's subject and subjectKeyId should return an empty result

File: integration_tests/cli/pki-remove-x509-certificates.sh [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+echo $result | jq
+check_response "$result" "\"subjectKeyId\": \"$root_cert_subject_key_id\""
+check_response "$result" "\"subjectKeyId\": \"$leaf_cert_subject_key_id\""
+check_response "$result" "\"serialNumber\": \"$root_cert_1_serial_number\""
+check_response "$result" "\"serialNumber\": \"$leaf_cert_serial_number\""
+response_does_not_contain "$result" "\"subjectKeyId\": \"$intermediate_cert_subject_key_id\""
+response_does_not_contain "$result" "\"serialNumber\": \"$intermediate_cert_2_serial_number\""
+response_does_not_contain "$result" "\"serialNumber\": \"$intermediate_cert_1_serial_number\""
+
+echo "Remove leaf certificate"
```
Comment (akarabashov) [inline integration_tests/cli/pki-remove-x509-certificates.sh:111]: It might be better to first revoke the leaf certificate and then proceed with the removal to ensure that the revoked certificate is also removed

File: integration_tests/grpc_rest/pki/helpers.go [hunk 4]
```diff
@@ -1818,4 +1818,98 @@ func Demo(suite *utils.TestSuite) {
...
+	}
+	_, err = suite.BuildAndBroadcastTx([]sdk.Msg{&msgRemoveX509Cert}, aliceName, aliceAccount)
+	require.NoError(suite.T, err)
+	// Check that it's removed from revoked list
+	_, err = GetRevokedX509Cert(suite, testconstants.IntermediateCertWithSameSubjectAndSKIDSubject, testconstants.IntermediateCertWithSameSubjectAndSKIDSubjectKeyID)
+	suite.AssertNotFound(err)
+	_, err = GetX509Cert(suite, testconstants.IntermediateCertWithSameSubjectAndSKIDSubject, testconstants.IntermediateCertWithSameSubjectAndSKIDSubjectKeyID)
+	suite.AssertNotFound(err)
+
+	// Remove x509 by subject, subject key id and serial number
```
Comment (akarabashov) [inline integration_tests/grpc_rest/pki/helpers.go:1846]: extra comment?

File: x/pki/handler_test.go
Comment (akarabashov) [inline x/pki/handler_test.go:1]: Can you please add tests for negative cases?
